packages/eslint-plugin/test/corpus.js

```diff
@@ -0,0 +1,155 @@
+module.exports = {
+  clearlyValidAsync: [
+    {
+      code: `const expr = async () => boo();`,
+    },
+    {
+      code: `const arrow = async (boo) => {
+        await boo();
+      }`,
+    },
+    {
+      code: `async function forAwaitOf() {
+        let b;
+        for await (b of baz()) {
+          await qux();
+        }
+      }`,
+    },
+    {
+      code: `async function forAwaitOfBy2() {
+        for await (const b of baz()) {
+          await qux();
+          for await (const c of crux()) {
+            await quux();
+          }
+          await quuz();
+        }
+      }`,
+    },
+    {
+      code: `async function awaitConst() {
+        await null;
+        if (baz()) {
+          await 3;
+        }
+      }`
+    },
+    {
+      code: `async function awaitFunc() {
+        await bar();
+        if (baz()) {
+          await qux();
+        }
+      }`,
+    },
+    {
+      code: `async function awaitTry() {
+        await bar();
+        try {
+          await zot();
+        } catch {
+          boo();
+        }
+      }`,
+    },
+    {
+      code: `async function awaitNestedTryCatch() {
+        await null;
+        try {
+          await zot();
+        } catch {
+          fn();
+        }
+      }`,
+    },
+    {
+      code: `async function awaitNestedTryFinally() {
+        await null;
+        try {
+          await zot();
+        } finally {
+          fn();
+        }
+      }`,
+    },
+    {
+      code: `async function awaitNestedTryCatchFinally() {
+        await null;
+        try {
+          zot();
+        } catch {
+          await boo();
+        } finally {
+          fn();
+        }
+      }`,
+    },
+    {
+      code: `const outer = async () => {
+        const inner = async () => {
+          await foo();
+        };
+        return inner;
+      }`
+    }
+  ],
+  subtlyAsync: [
```

Comment:
This organization of cases is a little confusing to me. It looks like `clearlyValidAsync` is just the `valid` case. And `invalid` is the union of`subtlyAsync` and the other cases in test-no-nested-await.js. Why are the cases split between files?

If you want two dimensions (validity and subtlety) please have 4 lists, all defined in this corpus. The test file should just load the cases from the corpus and run them.

Please also define the lists as constants to the export reads quickly:
```
module.exports = {
  valid,
  invalid,
};
```
or
```
module.exports = {
  clearlyValid,
  clearlyInvalid,
  subtelyValid,
  subtelyInvalid,
};
```
I'm on the fence about having the subtlety dimension at tall, as it's subjective. But it does seem worth calling out "you might be surprised by" cases for end user documentation. This serves that purpose.

Reply:
The `subtlyAsync` case is stuff that isn't valid for the `no-nested-await` rule, but _is_ valid for the `no-ambiguous-async-stack` rule that relies on control-flow analysis, but I haven't yet made a PR for (it is in my back pocket for completeness).  That's why it isn't labelled with "valid" or "invalid".

But you're right... the test cases should all live in `corpus.js` and have clear labels.